static/app/locale.tsx

```diff
@@ -269,30 +271,29 @@ export function renderTemplate(template: ParsedTemplate, components: ComponentMa
  * NOTE: This is a no-op and will return the node if LOCALE_DEBUG is not
  * currently enabled. See setLocaleDebug and toggleLocaleDebug.
  */
-function mark<T>(node: T): T {
+function mark(node: string | React.ReactNode): string {
   if (!LOCALE_DEBUG) {
-    return node;
+    return node as string;
   }
 
   // TODO(epurkhiser): Explain why we manually create a react node and assign
   // the toString function. This could likely also use better typing, but will
   // require some understanding of reacts internal types.
-  const proxy: any = {
+  const proxy: React.ReactNode = {
```

Comment:
are we sure about the `React.ReactNode` type? shouldn't it be like the following? 🤷‍♀️

```
{
    $$typeof: symbol;
    type: string;
    key: null;
    ref: null;
    props: {
        style: {
            background: string;
            outline: string;
        };
        children: React.ReactNodeArray;
    };
    _owner: null;
    _store: {};
}
```

Reply:
Yeah, tbh it doesnt really matter because we return `as string` in the end 😅. If I try typing it as Reac.ReactElement then I get an error from $$typeof...

Reply:
I think that even if in the end we return a `string`, it is important to define the correct type for `proxy`

Reply:
Yeah, correct and 100% agree. Let me see if I can make it work with React.CreateElement and if we can still override the toString, but tbh since I didn't find any usage, I'm kinda hoping we can remove it 😄

Reply:
So, I just tested and it will not work if we just try and override toString on React.createElement as the return object is frozen (expected), however we can reassign the component to a new object and change toString there... ([solution](https://codesandbox.io/s/jovial-wilson-6s9h8?file=/src/App.tsx))

I'm not a fan of this solution because we are adding runtime overhead of re-copying the object because our types are incorrect, so I wouldnt want to merge this. I think we are best leaving this as is for now even though it's not perfect and see if we can instead drop the whole proxy concept :)